src/results/mod.rs

```diff
@@ -138,22 +138,28 @@ impl Results {
         self.app_target_sdk = Some(sdk);
     }
 
-    pub fn add_vulnerability(&mut self, vuln: Vulnerability) {
-        match vuln.get_criticality() {
+    pub fn add_vulnerability(&mut self, config: &Config, vuln: Vulnerability) {
+        if ! match vuln.get_criticality() {
             Criticality::Warning => {
-                self.warnings.insert(vuln);
+                self.warnings.insert(vuln)
             }
             Criticality::Low => {
-                self.low.insert(vuln);
+                self.low.insert(vuln)
             }
             Criticality::Medium => {
-                self.medium.insert(vuln);
+                self.medium.insert(vuln)
             }
             Criticality::High => {
-                self.high.insert(vuln);
+                self.high.insert(vuln)
             }
             Criticality::Critical => {
-                self.critical.insert(vuln);
+                self.critical.insert(vuln)
+            }
+        } {
+            if config.is_verbose() {
+                println!("Failed to insert vulnerability into category, try again or open an issue");
+            } else {
+                println!("A rare issue occured. Please try again");
```

Comment:
This message does not really say much. This error should never happen, but if it does, maybe it's a good idea to write some proper error, or even panic (at least in debug mode, with a `debug_assert!`).

Reply:
Saying much more would kinda belong to verbose-mode in my opinion. Maybe something like "An internal error occurred. Please try again"  ?

Reply:
The thing is that this error should never happen, and if it happens, it depends on the input almost certainly, so trying again won't help. I would say something like `internal error happened, please, run with verbose mode to get more information`, or something along those lines.